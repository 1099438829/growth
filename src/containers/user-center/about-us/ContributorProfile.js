import React from 'react';
import { Linking, ScrollView, Text, View } from 'react-native';
import { List, ListItem } from 'react-native-elements';
import AppStyle from '../../../theme/styles';

const profileItemRenders = {
  github: value => (
    <ListItem
      key="github"
      title={`GitHub: @${value}`}
      rightIcon={{ name: 'github', type: 'font-awesome' }}
      onPress={() => Linking.openURL(`https://github.com/${value}`)}
    />
  ),
  weibo: value => (
    <ListItem
      key="weibo"
      title={`微博: @${value}`}
      rightIcon={{ name: 'weibo', type: 'font-awesome' }}
    />
  ),
  zhihu: value => (
    <ListItem
      key="zhihu"
      title={`知乎: @${value}`}
      rightIcon={{ name: 'quora', type: 'font-awesome' }}
    />
  ),
  blog: value => (
    <ListItem
      key="blog"
      title={`博客: ${value}`}
      rightIcon={{ name: 'rss', type: 'font-awesome' }}
      onPress={() => Linking.openURL(value)}
    />
  ),
};

const genProfileItems = (profile) => {
  const profileItemKeys = Object.keys(profileItemRenders);
  const availableItemKeys = profileItemKeys.filter(key => profile[key]);
  return availableItemKeys.map(key => profileItemRenders[key](profile[key]));
};

const ContributorProfile = props => (
  <ScrollView style={AppStyle.userCenterBasisStyle}>
    <View style={{ paddingLeft: 20, marginTop: 12, marginBottom: -6 }}>
      <Text style={{ fontSize: 14 }}>
        {props.contributor.duty}
      </Text>
    </View>
    <List>
      {genProfileItems(props.contributor.profile)}
    </List>
  </ScrollView>
);

ContributorProfile.propTypes = {
  // eslint-disable-next-line
  contributor: React.PropTypes.object.isRequired,
};

export default ContributorProfile;
