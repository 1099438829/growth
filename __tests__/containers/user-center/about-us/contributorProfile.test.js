import React from 'react';
// import { Linking } from 'react-native';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import ContributorProfile from '../../../../src/containers/user-center/about-us/ContributorProfile';

const contributor = {
  nickname: 'nickname',
  duty: 'duty',
  bio: 'bio',
  profile: {
    github: 'github',
    weibo: 'weibo',
    zhihu: 'zhihu',
    blog: 'http://blog.io',
  },
};

describe('test user center / about us / contributor profile', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ContributorProfile contributor={contributor} />,
    );

    const treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
  });
  //
  // it('test for pressing links', () => {
  //   const spy = jest.spyOn(Linking, 'openURL');
  //
  //   const wrapper = shallow(
  //     <ContributorProfile contributor={contributor} />,
  //   );
  //
  //   wrapper.find({ title: `GitHub: @${contributor.profile.github}` }).props().onPress();
  //   wrapper.find({ title: `博客: ${contributor.profile.blog}` }).props().onPress();
  //
  //   expect(spy).toHaveBeenCalledTimes(2);
  // });
});
