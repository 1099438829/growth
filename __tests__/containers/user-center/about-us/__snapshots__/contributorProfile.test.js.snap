// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test user center / about us / contributor profile renders correctly 1`] = `null`;
